templates/aws-stack.yml

```diff
@@ -811,6 +821,7 @@ Resources:
                   --==BOUNDARY==--
                 - {
                     LocalSecretsBucket: !If [ CreateSecretsBucket, !Ref ManagedSecretsBucket, !Ref SecretsBucket ],
+                    AgentTokenResolver: !If [ UseSSMAgentToken, !Sub "$(aws ssm get-parameter --name ${BuildkiteAgentTokenPath} --with-decryption | jq -r .Parameter.Value)", !Ref BuildkiteAgentToken ],
```

Comment:
I'd move this logic to the `bk-install-elastic-stack.sh` script, same deal as below, just blank this out if `UseSSMAgentToken` is true.